tutorials/shading/your_first_shader/what_are_shaders.rst

```diff
@@ -0,0 +1,111 @@
+.. _doc_what_are_shaders:
+
+What are shaders?
+=================
+
+Introduction
+------------
+
+So, you have decided to give shaders a try. You have likely heard that they can be used to 
+create interesting effects that run incredibly fast. You have also likely heard that they
+are terrifying. Both are true. 
+
+Shaders can be used to create a wide range of effects (in fact everything drawn in a modern
+rendering engine is done with shaders). 
+
+Writing shaders can also be very difficult for people unfamiliar with them. Godot tries to make writing
+shaders a little easier by exposing many useful built-in features and handling some of the
+lower-level initialization work for you. However, GLSL (the OpenGL Shading Language, which Godot uses)
+is still unintuitive and restricting, especially for users who are used to GDScript.
+
+But what are they?
+------------------
+
+Shaders are a special kind of program that runs on Graphics Processing Units (GPUs). All computers
+have some sort of GPU, either one integrated into their CPU or discrete (meaning it is a separate 
+hardware component). GPUs are especially useful for rendering because they are optimized for running
+thousands of instructions at once.
+
+The output of the shader is the colored pixels of the object drawn to the viewport. Shaders operate
+with a fixed-functionality pipeline. The standard process is the vertex -> fragment shader pipeline. The vertex
```

Comment:
1. Its fixed-function pipeline
2. Shaders were the evolution of the fixed-function pipeline. https://gamedevelopment.tutsplus.com/articles/the-end-of-fixed-function-rendering-pipelines-and-how-to-move-on--cms-21469

Reply:
I was using fixed-functionality to refer to the vertex -> fragment processing pipeline. Which is technically called the shader pipeline. But in a sense it is a pipeline with a fixed functionality, but not the fixed-function pipeline. Clearly this was confusing. I have tried to clear it up and use alternative language throughout.